### PR TITLE
test: update babel alignment tests to latest version

### PR DESCRIPTION
**Pull request by armano2**
## PR Checklist

-   [x] Steps in [CONTRIBUTING.md](https://github.com/typescript-eslint/typescript-eslint/blob/main/CONTRIBUTING.md) were taken

## Overview

this change is required for merge of #4692

### CODE REVIEW

File: packages/shared-fixtures/fixtures/typescript/basics/never-type-param.src.ts [hunk 1]
```diff
@@ -1,2 +1,2 @@
-const x: X<never>;
+var x: X<never>;
```
Comment (JoshuaKGoldberg) [inline packages/shared-fixtures/fixtures/typescript/basics/never-type-param.src.ts:1]: For my learning: why the change?

Comment (armano2) [reply packages/shared-fixtures/fixtures/typescript/basics/never-type-param.src.ts:1]: babel reports now errors for unidealized const variables, witch is tested in

```
    /**
     * [BABEL ERRORED, BUT TS-ESTREE DID NOT]
     * SyntaxError: Missing initializer in const declaration.
     */
    'var-with-definite-assignment',
```

**Review by JamesHenry — APPROVED** [review]
Love to see some of these ancient discrepancies have been resolved on the babel side 🎉
